ballerina/modules/soap11/soap11.bal

```diff
@@ -73,13 +83,18 @@ public isolated client class Client {
             lock {
                 wssec:OutboundSecurityConfig? outboundSecurity = self.outboundSecurity.clone();
                 do {
-                    if outboundSecurity is wssec:OutboundSecurityConfig {
-                        return check soap:applyOutboundConfig(outboundSecurity.clone(), response.clone());
+                    if outboundSecurity is wssec:OutboundSecurityConfig && outboundSecurity != {} {
+                        if response is xml {
+                            return check soap:applyOutboundConfig(outboundSecurity.clone(), response.clone(), false);
```

Comment:
```suggestion
                            return soap:applyOutboundConfig(outboundSecurity.clone(), response.clone(), false);
```

Reply:
When using error handling with `on fail`, we need to use `check` with `return`, in order to pass the error to `on fail` clause